frontend/utilities/campaign_helpers/index.ts

```diff
@@ -114,13 +122,21 @@ const updateCampaignStateFromStatus = (
 
 export const updateCampaignState = (socketData: ISocketData) => {
   return ({ campaign }: ICampaignState) => {
+    const { renderFlash } = useContext(NotificationContext);
     switch (socketData.type) {
       case "totals":
         return updateCampaignStateFromTotals(campaign, socketData);
       case "result":
         return updateCampaignStateFromResults(campaign, socketData);
       case "status":
         return updateCampaignStateFromStatus(campaign, socketData);
+      case "error":
+        if (socketData.data.includes("unexpected exit in receiveMessages")) {
+          renderFlash(
+            "error",
+            "Fleet's connection to Redis failed. If this issue persists, please contact your administrator."
+          );
```

Comment:
Agree, but to reduce risk in this first iteration I want to start with this one in particular.
Mainly because it's the one that will be logged when Fleet stops receiving results from Redis, so we would like the UI to display that something is off (currently in `main` nothing is displayed and it looks like the query is still running...)